source/text/implementation/vss-strings.adb

```diff
@@ -186,17 +186,36 @@ package body VSS.Strings is
         VSS.Strings.Iterators.Characters.Internals.First_Character (Self);
    end First_Character;
 
+   -------------
+   -- Handler --
+   -------------
+
+   function Handler
+     (Self : Virtual_String'Class)
+      return access Abstract_String_Handler'Class is
+   begin
+      if Self.Data.In_Place then
+         return VSS.Strings.Configuration.In_Place_Handler;
+
+      else
+         return Self.Data.Handler;
+      end if;
+   end Handler;
+
    --------------
    -- Is_Empty --
    --------------
 
    function Is_Empty (Self : Virtual_String'Class) return Boolean is
+      Handler : constant access Abstract_String_Handler'Class := Self.Handler;
+
    begin
-      return
-        (if Self.Data.In_Place
-         then VSS.Strings.Configuration.In_Place_Handler.Is_Empty (Self.Data)
-         else Self.Data.Handler = null
-           or else Self.Data.Handler.Is_Empty (Self.Data));
+      if Handler /= null then
```

Comment:
For such statements I prefer to swap then<-> else, so if-condition is "positive":
```ada
if Handler = null then
   return True;
else
   return Handler.Is_Empty (Self.Data);
end if;
```

See [Positive Forms of Logic](https://en.wikibooks.org/wiki/Ada_Style_Guide/Programming_Practices#Positive_Forms_of_Logic) in Ada Style Guide.